apps/dashboard/src/app/(dashboard)/published-contract/[publisher]/[contract_id]/utils/getPublishedContractPageMetadata.ts

```diff
@@ -0,0 +1,49 @@
+import { format } from "date-fns/format";
+import type { Metadata } from "next";
+import { correctAndUniqueLicenses } from "../../../../../../lib/licenses";
+import { PublishedContractOG } from "../../../../../../og-lib/url-utils";
+import { getPublishedContractsWithPublisherMapping } from "./getPublishedContractsWithPublisherMapping";
+
+export async function getPublishedContractPageMetadata(params: {
+  publisher: string;
+  contract_id: string;
+  pageType: "view" | "deploy";
+}): Promise<Metadata> {
+  const { publisher, contract_id, pageType } = params;
+
+  const publishedContracts = await getPublishedContractsWithPublisherMapping({
+    publisher: publisher,
+    contract_id: contract_id,
+  });
+
+  const publishedContract = publishedContracts[0];
+
+  const publishedContractName =
+    publishedContract?.displayName || publishedContract?.name;
+
+  const ogImageURL = PublishedContractOG.toUrl({
+    name: `${pageType === "deploy" ? "Deploy " : ""}${publishedContractName}`,
+    description: publishedContract.description,
+    version: publishedContract.version || "latest",
+    publisher: publisher,
+    license: correctAndUniqueLicenses(publishedContract.licenses),
+    publishDate: format(
+      new Date(
+        Number.parseInt(publishedContract.publishTimestamp.toString() || "0") *
+          1000,
+      ),
+      "MMM dd, yyyy",
+    ),
+    logo: publishedContract.logo,
+    // extension: extensionNames, TODO, but not a blocker
+    // publisherAvatar:  TODO, but not a blocker
+  });
```

Comment:
this should instead use https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image#generate-images-using-code-js-ts-tsx

Reply:
I used this method on portal - It was VERY slow and I had to change it and replace it with this method